src/modules/supplementalPositionMethods.ts

```diff
@@ -177,4 +184,132 @@ export const supplementalPositionMethods = {
             position.jobId && (await jobDelete(position.jobId));
         }
     },
+
+    createRequest: async (data: CreateSupplementalPositionRequest, sessionUserId: string) => {
+        const startSupplementalPositionDate = data.supplementalPositions[0].workStartDate;
+        if (!startSupplementalPositionDate) {
+            throw new TRPCError({
+                code: 'BAD_REQUEST',
+                message: `No workStartDate in creation request for supplemental position for user with id ${data.userTargetId}`,
+            });
+        }
+
+        startSupplementalPositionDate.setUTCHours(config.employmentUtcHour);
+
+        const user = await userMethods.getById(data.userTargetId);
+
+        if (!user) {
+            throw new TRPCError({ code: 'NOT_FOUND', message: `No user with id ${data.userTargetId} found` });
+        }
+
+        if (!user.login) {
+            throw new TRPCError({ code: 'NOT_FOUND', message: `User with id ${data.userTargetId} has no login` });
+        }
+
+        const name = trimAndJoin([data.surname, data.firstName, data.middleName]);
+
+        await serviceMethods.updateUserServicesInRequest({
+            userId: user.id,
+            services: user.services,
+            phone: data.phone,
+            personalEmail: data.personalEmail,
+            workEmail: data.workEmail,
+        });
+
+        const requestValues: InsertExpression<DB, 'UserCreationRequest'> = {
+            type: data.type,
+            userTargetId: data.userTargetId,
+            supervisorId: data.supervisorId,
+            organizationUnitId: data.organizationUnitId,
+            email: user.email,
+            login: user.login,
+            createExternalAccount: false,
+            name,
+            creatorId: sessionUserId,
+            groupId: data.groupId,
+            title: data.title,
+            corporateEmail: data.corporateEmail,
+            comment: data.comment,
+            workEmail: data.workEmail,
+            personalEmail: data.personalEmail,
+            services: [],
+            workMode: data.workMode,
+            workModeComment: data.workModeComment,
+            equipment: data.equipment,
+            extraEquipment: data.extraEquipment,
+            workSpace: data.workSpace,
+            location: data.location,
+            unitId: data.unitId,
+            percentage: data.percentage ? data.percentage * percentageMultiply : undefined,
+        };
+
+        if (data.buddyId) {
+            await userMethods.getByIdOrThrow(data.buddyId);
+            requestValues.buddyId = data.buddyId;
+        }
+
+        if (data.recruiterId) {
+            await userMethods.getByIdOrThrow(data.recruiterId);
+            requestValues.recruiterId = data.recruiterId;
+        }
+
+        const request = await db
```

Comment:
`From this line to 313 line`

If one of chain requests will throw error then partially create/update entities in several tables and will not restored previous values

Needs wrap all update/insert derectives into transaction